apis/apiextensions/v1/composition_transforms.go

```diff
@@ -65,6 +74,157 @@ type Transform struct {
 	Convert *ConvertTransform `json:"convert,omitempty"`
 }
 
+// Validate this Transform is valid.
+//
+//nolint:gocyclo // This is a long but simple/same-y switch.
+func (t *Transform) Validate() *field.Error {
+	// TODO(phisco): we should validate each transform type is valid too, e.g. TransformTypeString => String defines the required field for it's type
```

Comment:
Is this `todo` still valid?

Reply:
yes, addressed [here](https://github.com/crossplane/crossplane/pull/3921/commits/77cb4bb335c40b764f2ccd8ac5fe02cbd5ed511b)